src/modi/mod.rs

```diff
@@ -0,0 +1,210 @@
+use std::mem;
+
+use crate::common::*;
+use crate::dbi::Module;
+use crate::msf::Stream;
+use crate::symbol::SymbolIter;
+use crate::FallibleIterator;
+
+mod c13;
+mod constants;
+
+#[derive(Clone, Copy, Debug)]
+enum LinesSize {
+    C11(usize),
+    C13(usize),
+}
+
+/// This struct contains data about a single module from its module info stream.
+///
+/// The module info stream is where private symbols and line info is stored.
+pub struct ModuleInfo<'s> {
+    stream: Stream<'s>,
+    symbols_size: usize,
+    lines_size: LinesSize,
+}
+
+impl<'s> ModuleInfo<'s> {
+    /// Parses a `ModuleInfo` from it's Module info stream data.
+    pub(crate) fn parse(stream: Stream<'s>, module: &Module<'_>) -> Result<Self> {
+        let info = module.info();
+
+        let mut buf = stream.parse_buffer();
+        if buf.parse_u32()? != constants::CV_SIGNATURE_C13 {
+            return Err(Error::UnimplementedFeature(
+                "Unsupported module info format",
+            ));
+        }
+
+        let lines_size = if info.lines_size > 0 {
+            LinesSize::C11(info.lines_size as usize)
+        } else {
+            LinesSize::C13(info.c13_lines_size as usize)
+        };
+
+        let symbols_size = info.symbols_size as usize;
+        Ok(ModuleInfo {
+            stream,
+            symbols_size,
+            lines_size,
+        })
+    }
+
+    /// Get an iterator over the all symbols in this module.
+    pub fn symbols(&self) -> Result<SymbolIter<'_>> {
+        let mut buf = self.stream.parse_buffer();
+        buf.parse_u32()?;
```

Comment:
Excellent question, that was already there before and I didn't touch it. I **assume** it might be the PDB version signature, specifically `CV_SIGNATURE_C13`. Let's see if it's easy to dig up.

Reply:
Okay, it is in fact the CV signature. Since it's read when parsing the module already, it can simply be skipped and there is no need to check it again.